pyccel/codegen/printing/fcode.py

```diff
@@ -923,16 +923,16 @@ def _print_MacroCount(self, expr):
             shape = []
             for (s, i) in zip(_shape, var.indices):
                 if isinstance(i, Slice):
-                    if i.start is None and i.end is None:
+                    if i.start is None and i.stop is None:
                         shape.append(s)
                     elif i.start is None:
-                        if (isinstance(i.end, (int, LiteralInteger)) and i.end>0) or not(isinstance(i.end, (int, LiteralInteger))):
-                            shape.append(i.end)
-                    elif i.end is None:
+                        if (isinstance(i.stop, (int, LiteralInteger)) and i.stop>0) or not(isinstance(i.stop, (int, LiteralInteger))):
+                            shape.append(i.stop)
+                    elif i.stop is None:
                         if (isinstance(i.start, (int, LiteralInteger)) and i.start<s-1) or not(isinstance(i.start, (int, LiteralInteger))):
-                            shape.append(s-i.start)
+                            shape.append(s - i.start)
                     else:
-                        shape.append(i.end-i.start+1)
+                        shape.append(i.stop - i.start + 1)
```

Comment:
This code isn't tested (see https://app.codacy.com/gh/pyccel/pyccel/file/51246314423/coverage?bid=15415567&fileBranchId=15415567#l901)
and I'm not sure if it's used so it's not a huge problem, but this code is definitely wrong. Would you mind putting in the missing `PyccelMinus`, `PyccelAdd`, and `LiteralInteger` calls?

Reply:
Thanks :+1: